docs/device-api.md

```diff
@@ -0,0 +1,124 @@
+|                           | show_level            |    O     |       O        |      O       |     O      |      O       |
+|                           | show_log_origin       |    O     |       O        |      O       |     O      |      O       |
+| `MachineConfiguration`    | cpu_template          |    O     |       O        |      O       |     O      |      O       |
+|                           | ht_enabled            |    O     |       O        |      O       |     O      |      O       |
+|                           | mem_size_mib          |    O     |       O        |      O       |     O      |      O       |
+|                           | track_dirty_pages     |    O     |       O        |      O       |     O      |      O       |
+|                           | vcpu_count            |    O     |       O        |      O       |     O      |      O       |
+| `Metrics`                 | metrics_path          |    O     |       O        |      O       |     O      |      O       |
+| `MmdsConfig`              | ipv4_address          |    O     |       O        |      O       |   **R**    |      O       |
+| `NetworkInterface`        | allow_mmds_requests   |    O     |       O        |      O       |   **R**    |      O       |
+|                           | guest_mac             |    O     |       O        |      O       |   **R**    |      O       |
+|                           | host_dev_name         |    O     |       O        |      O       |   **R**    |      O       |
+|                           | iface_id              |    O     |       O        |      O       |   **R**    |      O       |
+|                           | rx_rate_limiter       |    O     |       O        |      O       |   **R**    |      O       |
+|                           | tx_rate_limiter       |    O     |       O        |      O       |   **R**    |      O       |
+| `PartialDrive`            | drive_id              |    O     |       O        |    **R**     |     O      |      O       |
+|                           | path_on_host          |    O     |       O        |    **R**     |     O      |      O       |
+| `PartialNetworkInterface` | iface_id              |    O     |       O        |      O       |   **R**    |      O       |
+|                           | rx_rate_limiter       |    O     |       O        |      O       |   **R**    |      O       |
+|                           | tx_rate_limiter       |    O     |       O        |      O       |   **R**    |      O       |
+| `RateLimiter`             | bandwidth             |    O     |       O        |    **R**     |   **R**    |      O       |
+|                           | ops                   |    O     |       O        |    **R**     |   **R**    |      O       |
+| `TokenBucket`             | one_time_burst        |    O     |       O        |    **R**     |   **R**    |      O       |
```

Comment:
The same case as with the "RateLimiter".

Reply:
Again if this is not an API request it should be remove.  If it is an API request, whether standalone or not, it should stay.

If it is an API request do you agree breaking it into two rows as for RateLimiter communicates the fact either or both devices are required?